scala/README.md

```diff
@@ -245,33 +278,41 @@ It is definitely recommended to read the full Twitter's "[Effective Scala]" guid
 
 ### Additional recommendations
 
-0. [Error handling](http://tersesystems.com/2012/12/27/error-handling-in-scala/).
+0. Avoid [pattern matching on Options](http://blog.originate.com/blog/2014/06/15/idiomatic-scala-your-options-do-not-match/).
+
+0. Never use `return`: [http://tpolecat.github.io/2014/05/09/return.html](http://tpolecat.github.io/2014/05/09/return.html)
 
-0. Using [Options](http://blog.originate.com/blog/2014/06/15/idiomatic-scala-your-options-do-not-match/).
+0. Parentheses in Scala are not optional. Be aware of extraneous parentheses:
 
-0. Do not use `return`: http://tpolecat.github.io/2014/05/09/return.html
+    ```scala
+    val bad = Set.empty() // expands to Set.empty.apply(), equals to Boolean false
+
+    val evil = Seq(1,2,3).toSet() // same as above
+    ```
 
 0. Always use the most generic collection type possible, typically one of: `Iterable[T]`, `Seq[T]`, `Set[T]`, or `Map[T]`.
 
 0. Use `Seq[T]`, not `List[T]` (see: [http://stackoverflow.com/a/10866807/410286](http://stackoverflow.com/a/10866807/410286)) except where you specifically need to force one implementation over another. The most common exception is that Play form mappers require `List[T]`, so you have to use it there. `Seq` is the interface, `List` the implementation, analogous to `Map` and `HashMap` in Java.
 
-0. Do not overuse tuples, decompose them or better, use case classes:
+0. Do not overuse tuples, decompose them or use case classes:
```

Comment:
I preferred the emphasis on using case classes. If I see code like `def name(person: (String, String)): String`, I immediately suspect that `(String, String)` will grow to embody the rest of a `Person` one day. Encouraging that naming earlier rather than later is better.